### PR TITLE
update formatting throughout codebase

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description

- *Category*:  documentation
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5243

### Changes and notes
 We previously settled on sticking with numpy style documentation.
This PR makes small tweaks for consistency. Most notably, it implements
a new type hints extension so that we do not need to duplicate
type hints in the signature as well as the docstring.

### CODE REVIEW

File: src/vivarium/framework/engine.py [hunk 1]
```diff
@@ -358,33 +357,39 @@ class Builder:
 
     Attributes
     ----------
-    logging: LoggingInterface
+    configuration : ``LayeredConfigTree``
```
Comment (stevebachmeier) [inline src/vivarium/framework/engine.py:360]: I couldn't get this to work so I just kept it nit-picked. Quotes didn't work but for some reason double back-ticks did 🤷‍♂️

File: docs/source/conf.py [hunk 2]
```diff
@@ -50,6 +50,7 @@
 # ones.
 extensions = [
     "sphinx.ext.autodoc",
+    "sphinx_autodoc_typehints",
```
Comment (stevebachmeier) [inline docs/source/conf.py:53]: This is the thing that allows us to not type hint parameters and returns in the docstring even though we already have them typed in the signature.

**Review by albrja — APPROVED** [review]

File: src/vivarium/component.py [hunk 3]
```diff
@@ -159,54 +158,47 @@ def name(self) -> str:
 
     @property
     def sub_components(self) -> List["Component"]:
-        """
-        Provide components managed by this component.
+        """Provide components managed by this component.
 
         Returns
         -------
         List[Component]
```
Comment (rmudambi) [inline src/vivarium/component.py:165]: Wasn't this supposed to be removed?

**Review by rmudambi — APPROVED** [review]

File: src/vivarium/component.py (hunk 3 above)
Comment (stevebachmeier) [reply src/vivarium/component.py:165]: This (and the `str` return type above it) were the two I reffered to earlier that for some reason sphinx has a problem with removing.

**Pull request merged**